### ISSUE TITLE
Specific accuracy of llama models

### ISSUE THREAD

**Issue by IQ179** [nelson-liu/lost-in-the-middle#13]
Hello, I saw a graph showing the results of llama2 models (foundation model and chat model) in the paper.
Could you tell me the specific accuracy of them?

**Comment by nelson-liu** [comment]
Here's a JSON object with the results of the LLaMA-2 models. keys are model names, value is a list of tuples where each tuple is (gold position, accuracy). Let me know if you need anything else!

```
{
    "Llama-2-7b-chat-hf": [
        [
            0,
            0.428842504743833
        ],
        [
            4,
            0.41783681214421253
        ],
        [
            9,
            0.4459203036053131
        ],
        [
            14,
            0.45692599620493357
        ],
        [
            19,
            0.5540796963946869
        ]
    ],
    "Llama-2-13b-chat-hf": [
        [
            0,
            0.5149905123339659
        ],
        [
            4,
            0.4774193548387097
        ],
        [
            9,
            0.47666034155597725
        ],
        [
            14,
            0.5244781783681215
        ],
        [
            19,
            0.5882352941176471
        ]
    ],
    "Llama-2-70b-chat-hf": [
        [
            0,
            0.573055028462998
        ],
        [
            4,
            0.538899430740038
        ],
        [
            9,
            0.5472485768500949
        ],
        [
            14,
            0.6003795066413662
        ],
        [
            19,
            0.6998102466793169
        ]
    ],
    "Llama-2-7b-hf": [
        [
            0,
            0.23440453686200377
        ],
        [
            4,
            0.23100189035916824
        ],
        [
            9,
            0.23553875236294897
        ],
        [
            14,
            0.24914933837429112
        ],
        [
            19,
            0.410586011342155
        ]
    ],
    "Llama-2-13b-hf": [
        [
            0,
            0.43629489603024574
        ],
        [
            4,
            0.2166351606805293
        ],
        [
            9,
            0.21323251417769376
        ],
        [
            14,
            0.23251417769376181
        ],
        [
            19,
            0.40869565217391307
        ]
    ],
    "Llama-2-70b-hf": [
        [
            0,
            0.499054820415879
        ],
        [
            4,
            0.4170132325141777
        ],
        [
            9,
            0.41776937618147447
        ],
        [
            14,
            0.4487712665406427
        ],
        [
            19,
            0.5705103969754254
        ]
    ]
}
```

**Comment by aryopg** [comment]
Hi, amazing work @nelson-liu as always!
A noob question: I was just wondering about the postprocessing of the generated answer. If I understand your code correctly, you basically took the raw output directly. I tried generating answers for NQ open using LLaMA2 7b chat model and it tends to be a bit verbose in its generation, starting the answer with a sentence like "Based on the provided search results, the ...". What strategy did you use to mitigate this?

**Comment by nelson-liu** [comment]
Hi @aryopg , thanks for the kind words! Yes, we took the LLaMA-2 chat model outputs directly. The chat models are indeed quite verbose---this ends up making EM / F1 a meaningless metric, but there's still some signal in **best substring EM** (i.e., do any of the gold answer strings show up in the generation), especially if the goal is to compare performance of changing the answer-bearing paragraph (this paper is somewhat relevant in this regard https://arxiv.org/abs/2305.06984 ), since the filler / boilerplate text generated by the chat model basically never contains the answer. I think the relative comparisons between settings is very trustworthy, but it's quite possible that the absolute performance %s differ from human eval by a few percentage points (but this paper is mostly concerned with relative comparisons anyway, i.e., start vs. middle vs. end).

This issue above is primarily why we used best substring EM as our metric of choice. In a perfect world, I would have liked to do some sort of model-based eval vs. just relying on string overlap, but from manually looking at the results I didn't see many where it would have made a huge difference.

I think this becomes a little more of a problem when you have models that are _very_ verbose in ways that aren't just boilerplate (e.g., consider a model that repeats the input text before answering---copying the input to the output would result in perfect best substring EM). We didn't observe this in any of the models that we tested, but it's possible that future models might require something a bit more sophisticated than best substring EM. Picking the right metric for the job can ultimately depend on looking at model behavior.

Anyway, I realize this is a little bit of a brain-dump---let me know if this answered your question / if you have any follow-up questions.

**Comment by aryopg** [comment]
Hi @nelson-liu , thank you very much for the answer! It makes total sense, and we're also planning to use best substring EM as that would be the fairest metric in this case.  I can indeed see a very verbose model, particularly if you prompt it to do exactly that (e.g. CoT and all that jazz). Probably we should use/introduce metrics that are independent of this. Wonder if something like BERTScore (with some tweaks) would help here.

Keep the brain-dump coming tho! In the constraint of my question, you've sufficiently answered it! We can move the discussion to emails instead :)

**Comment by nelson-liu** [comment]
> I can indeed see a very verbose model, particularly if you prompt it to do exactly that (e.g. CoT and all that jazz). Probably we should use/introduce metrics that are independent of this. Wonder if something like BERTScore (with some tweaks) would help here.

Yeah, I can see something like that working reasonably, but there are still cases where you need to actually understand the model response. In some sense, evaluating the responses is just another QA task, one where best subspan em seems to correlate quite well on (at least for some models).

Part of me feels like the most robust solution would probably be just eval with a LM (e.g., `gpt-4-turbo` or some fine-tuned open model). you'd just give it:

1. the question
2. the gold answer
3. the model response

and ask it to assess whether the model response accurately conveys the gold answer. we tried this for a bit with gpt-4 and it worked quite well,l but i ended up not doing it for the paper results because it was unnecessary cost and I didn't think was really telling us anything that best subspan EM wouldn't. If i were redoing this project today and had a model that was more verbose (e.g., claude 2.1 seems to really enjoy rephrasing its input text, maybe because it's trained with some sort of context distillation), I think the LM eval is the first thing I'd try.
